### ISSUE TITLE
Sync text date format not match system settings

### ISSUE THREAD

**Issue by MDXDave** [bitwarden/android#3717]
### Bitwarden Beta

- [ ] I'm using the new native Bitwarden Beta app and I'm aware that legacy .NET app bugs should be reported in [bitwarden/mobile](https://github.com/bitwarden/mobile)

### Steps To Reproduce

1. Change device languages to german 
2. Go to 'Einstellungen' 
3. Click on 'Sonstiges'

### Expected Result

The text should show the last synced date & time in the system settings (e.g. DD/MM/YYY, HH:MM, 24 hours): 

> Zuletzt synchronisiert 12/08/2024 19:40 

### Actual Result

The text shows a (fixed?) Format, not reespecting the system settings (M/DD/YYYY, 12 hours) 
> Zuletzt synchronisiert: 8/12/2024 7:40 PM

### Screenshots or Videos

![Screenshot_20240812_194223_Bitwarden Beta](https://github.com/user-attachments/assets/e14b3150-07b4-443a-b5e1-c4d879fdb4d1)

### Build Version

2024.08 18985

### Environment Details

- Device: Samsung Galaxy S23 Ultra
- OS: Android 14

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by richardb64** [comment]
I've just noticed that this is even an issue between dialects of English - the "Last updated" message in the Android app appears to be fixed to display in the confusing half-reversed American format (MM/dd/yy h:mm a).

If for some technical reason it's really not possible for this to be properly localised, could it perhaps at least then universally use the ISO standard format? (yyyy-MM-dd HH:mm)?

**Comment by edent** [comment]
This error also shows up in the TOTP section. This makes it misleading when trying to understand if a TOTP code is valid.

I think the issue is with this line:

https://github.com/bitwarden/android/blob/61c48bf6736cb6777336ec1b9e9b30ff5b7102e8/app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/settings/other/OtherViewModel.kt#L28

However, there seem to be a few places where "month / day" is hardcoded 

https://github.com/search?q=repo%3Abitwarden%2Fandroid+%22d%2Fyy%22&type=code

**Comment by richardb64** [comment]
Any progress on this?  ETA for a fix?  Is there anywhere that users can track any further discussion?  This is really quite a serious and basic fault in an application that desires global reach.  For anyone anywhere outside the US, the "M/d/yyyy" format is dangerously misleading.

**Comment by richardb64** [comment]
This general problem isn't specific to Android, either.  The Windows desktop app similarly ignores the user's locale and displays dates in the form "Jan 31, 2020, 1:23:45 PM".  At least the use of a textual month makes it less ambigious, but it's still **wrong**.  The web vault seems to get it right.

**Comment by richardb64** [comment]
Also to be clear (because the related issues have been closed as duplicates): this issue doesn't _only_ affect the "last sync" date.  It seems to affect pretty much everywhere a date is displayed across the entire application - notably also in the "last updated" date on individual records.

**Comment by david-livefront** [comment]
@richardb64 This issue should have been fixed [here](https://github.com/bitwarden/android/pull/5333).

Can you confirm that this issue is resolved for you?

**Comment by richardb64** [comment]
Looks good!  Many thanks for getting this over the line @david-livefront!

The Windows desktop app still seems to have the same problem, though - do you know if that's being worked on (with I guess an almost identical fix to the one you've successfully applied to Android)?

**Comment by david-livefront** [comment]
@richardb64 I don't believe anyone on the desktop app is investigating this. I would recommend writing up an issue in the [clients repo](https://github.com/bitwarden/clients).

**Comment by david-livefront** [comment]
@richardb64 I did reach out to one of the client devs and we were hoping you could provide a screenshot of exactly where this issue is occurring on desktop?

**Comment by pamperer562580892423** [comment]
As I see it, at least the browser extension 2025.10.0 still has it (Settings --> Vault options):

<img width="613" height="114" alt="Image" src="https://github.com/user-attachments/assets/19ea213f-9a7f-40ff-abac-49ef641d8c2b" />

PS: ... as I also documented it here: https://github.com/bitwarden/android/issues/5327 😉

**Comment by richardb64** [comment]
> @richardb64 I did reach out to one of the client devs and we were hoping you could provide a screenshot of exactly where this issue is occurring on desktop?

No problem - I'll gather some examples from the desktop app.  Is there a relevant issue over on the other repo to attach them to?

**Comment by david-livefront** [comment]
> > [@richardb64](https://github.com/richardb64) I did reach out to one of the client devs and we were hoping you could provide a screenshot of exactly where this issue is occurring on desktop?
> 
> No problem - I'll gather some examples from the desktop app. Is there a relevant issue over on the other repo to attach them to?

I don't believe there is an issue for this at this time.

**Comment by david-livefront** [comment]
This issue has been resolved in the Android app for quite some time. Closing the issue.
